src/TwigComponent/src/ComponentTemplateFinder.php

```diff
@@ -15,33 +15,53 @@
 
 /**
  * @author Matheo Daninos <matheo.daninos@gmail.com>
+ *
+ * @internal
  */
 final class ComponentTemplateFinder implements ComponentTemplateFinderInterface
 {
+    private readonly ?string $directory;
+
     public function __construct(
-        private Environment $environment
+        private Environment $environment,
+        string $directory = null,
     ) {
+        $this->directory = \is_string($directory) ? (rtrim($directory, '/').'/') : null;
     }
 
     public function findAnonymousComponentTemplate(string $name): ?string
     {
         $loader = $this->environment->getLoader();
         $componentPath = rtrim(str_replace(':', '/', $name));
 
-        if ($loader->exists('components/'.$componentPath.'.html.twig')) {
-            return 'components/'.$componentPath.'.html.twig';
-        }
+        if (null === $this->directory) {
+            if ($loader->exists('components/'.$componentPath.'.html.twig')) {
+                return 'components/'.$componentPath.'.html.twig';
+            }
+
+            if ($loader->exists($componentPath.'.html.twig')) {
+                return $componentPath.'.html.twig';
+            }
+
+            if ($loader->exists('components/'.$componentPath)) {
+                return 'components/'.$componentPath;
+            }
+
+            if ($loader->exists($componentPath)) {
+                return $componentPath;
+            }
 
-        if ($loader->exists($componentPath.'.html.twig')) {
-            return $componentPath.'.html.twig';
+            return null;
         }
 
-        if ($loader->exists('components/'.$componentPath)) {
-            return 'components/'.$componentPath;
+        $template = $this->directory.$componentPath.'.html.twig';
+        if ($loader->exists($template)) {
+            return $template;
         }
 
-        if ($loader->exists($componentPath)) {
-            return $componentPath;
+        $template = $this->directory.$componentPath;
+        if ($loader->exists($template)) {
+            return $template;
```

Comment:
We could remove this now. Basically, the old, legacy logic is in the if statement above, unchanged. 👍 . So down here, we can do what we want. Does it make sense to have a template name that exactly matches the component name? Would that look like `<twig:foo.html.twig>`? idk...